Hearthstone Deck Tracker/Controls/Overlay/BobsBuddyPanel.xaml.cs

```diff
@@ -170,42 +227,134 @@ public Visibility InfoVisibility
+			SetAverageDamage(possibleResults);
 			WinRateDisplay = string.Format("{0:0.#%}", winRate);
 			TieRateDisplay = string.Format("{0:0.#%}", tieRate);
 			LossRateDisplay = string.Format("{0:0.#%}", lossRate);
 			PlayerLethalDisplay = string.Format("{0:0.#%}", playerLethal);
 			OpponentLethalDisplay = string.Format("{0:0.#%}", opponentLethal);
 
-			PlayerLethalOpacity = playerLethal > 0 ? 1 : 0.3;
-			OpponentLethalOpacity = opponentLethal > 0 ? 1 : 0.3;
+			PlayerLethalOpacity = playerLethal > 0 ? 1 : SoftLabelOpacity;
+			OpponentLethalOpacity = opponentLethal > 0 ? 1 : SoftLabelOpacity;
+
+			PlayerAverageDamageOpacity = possibleResults.Where(x => x > 0).Any() ? 1 : SoftLabelOpacity;
+			OpponentAverageDamageOpacity = possibleResults.Where(x => x < 0).Any() ? 1 : SoftLabelOpacity;
+		}
+
+
+		internal void SetLastOutcome(int lastOutcome)
+		{
+			_lastCombatResult = lastOutcome;
+			CheckIfDamageOutcomeOutsideEightyPercent();
+		}
+
+
+		private void CheckIfDamageOutcomeOutsideEightyPercent()
+		{
+			if(Config.Instance.SeenBobsBuddyAverageDamage)
+				return;
+			if(_lastCombatResult < 0 && _opponentDamageDealtBounds != null)
+			{
+				if(_lastCombatResult < _opponentDamageDealtBounds[0] || _lastCombatResult > _opponentDamageDealtBounds[1])
+				{
+					AttemptToExpandAverageDamagePanels(true);
+				}
+			}
+			else if(_lastCombatResult > 0 && _playerDamageDealtBounds != null)
+			{
+				if(_lastCombatResult < _playerDamageDealtBounds[0] || _lastCombatResult > _playerDamageDealtBounds[1])
+				{
+					AttemptToExpandAverageDamagePanels(true);
+				}
+			}
+		}
+
+		internal void SetAverageDamage(List<int> possibleResults)
+		{
+			var playerDamageDealtPossibilities = possibleResults.Where(x => x > 0).ToList();
+			var opponentSortedDamageDealtPossibilites = possibleResults.Where(x => x < 0).Select(y => y * -1).ToList();
+			opponentSortedDamageDealtPossibilites.Sort((x, y) => x.CompareTo(y));
+
+			_playerDamageDealtBounds = GetTwentiethAndEightiethPercentileFor(playerDamageDealtPossibilities);
+			_opponentDamageDealtBounds = GetTwentiethAndEightiethPercentileFor(opponentSortedDamageDealtPossibilites);
+
+			PlayerAverageDamageOpacity = _playerDamageDealtBounds == null ? SoftLabelOpacity : 1;
+			OpponentAverageDamageOpacity = _opponentDamageDealtBounds == null ? SoftLabelOpacity : 1;
+
+			AverageDamageGivenDisplay = FormatDamageBoundsFrom(_playerDamageDealtBounds);
+			AverageDamageTakenDisplay = FormatDamageBoundsFrom(_opponentDamageDealtBounds);
 		}
 
+		private List<int> GetTwentiethAndEightiethPercentileFor(List<int> possibleResults)
+		{
+			var count = possibleResults.Count;
+			if(count == 0)
+				return null;
+			return new List<int>() { possibleResults[(int)Math.Floor(.2 * count)], possibleResults[(int)Math.Floor(.8 * count)] };
+		}
+
+		private string FormatDamageBoundsFrom(List<int> from) => from == null ? "0" : from[0] == from[1] ? from[0].ToString() : string.Format("{0}–{1}", from[0], from[1]);
```

Comment:
This can be simplified as
```suggestion
		private string FormatDamageBoundsFrom(List<int> from) => from == null ? "0" : string.Join("–", from.Distinct()));
```

`string.Join` will omit the separator if there is only one item. So `string.Join("–", [1])` => `1` and `string.Join("–", [1, 2])` => `1–2`